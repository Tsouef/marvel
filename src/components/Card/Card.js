import React from 'react';
import { Link } from 'react-router-dom';
import './Card.css';

const Card = ({ informations, title, onClick, category }) => {
  const { thumbnail, id, description } = informations;
  return (
    <div className="card">
      <div className="card-image">
        <img alt="" src={`${thumbnail.path}.${thumbnail.extension}`} />
        <a
          onClick={onClick}
          className="btn-floating halfway-fab waves-effect waves-light red"
        >
          <i className="material-icons">add</i>
        </a>
      </div>
      <div className="card-content">
        <span className="card-title">{title}</span>
        <p dangerouslySetInnerHTML={{ __html: description }} />
      </div>
      <div className="card-action">
        <Link to={`/${category}/id/${id}`}>En voir plus</Link>
      </div>
    </div>
  );
};

export default Card;
