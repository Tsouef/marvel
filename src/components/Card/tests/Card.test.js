import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import React from 'react';
import Card from './Card';

describe('component <Card />', () => {
  it('should render Card component', () => {
    const props = {
      thumbnail: {
        path: 'http://i.annihil.us/u/prod/marvel/i/mg/5/90/4c4e014aa3086',
        extions: 'jpg'
      },
      id: '1',
      description: 'This is a description'
    };
    const tree = renderer
      .create(
        <MemoryRouter>
          <Card key="1" informations={props} title="Title" category="comics" />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
