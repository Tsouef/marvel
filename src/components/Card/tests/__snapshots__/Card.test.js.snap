// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Card /> should render Card component 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <img
      alt=""
      src="http://i.annihil.us/u/prod/marvel/i/mg/5/90/4c4e014aa3086.undefined"
    />
    <a
      className="btn-floating halfway-fab waves-effect waves-light red"
      onClick={undefined}
    >
      <i
        className="material-icons"
      >
        add
      </i>
    </a>
  </div>
  <div
    className="card-content"
  >
    <span
      className="card-title"
    >
      Title
    </span>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "This is a description",
        }
      }
    />
  </div>
  <div
    className="card-action"
  >
    <a
      href="/comics/id/1"
      onClick={[Function]}
    >
      En voir plus
    </a>
  </div>
</div>
`;
